src/mbgl/renderer/sources/render_raster_dem_source.cpp

```diff
@@ -26,18 +27,21 @@ void RenderRasterDEMSource::update(Immutable<style::Source::Impl> baseImpl_,
 
     enabled = needsRendering;
 
-    optional<Tileset> _tileset = impl().getTileset();
-
-    if (tileset != _tileset) {
-        tileset = _tileset;
+    const optional<Tileset>& implTileset = impl().tileset;
```

Comment:
Can this logic be pulled up to the base class?

Reply:
I thought of it and did not come to an elegant solution :thinking:  maybe just some aux protected method could help.. the problem is that not all tiled sources use tileset

Reply:
Done in the latest commit